src/collectors/loggly/test/testloggly.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/python
+# coding=utf-8
+##########################################################################
+
+import os
+import time
+import mock
+from test import CollectorTestCase
+from test import get_collector_config
+from test import unittest
+from test import run_only
+from mock import patch
+
+from diamond.collector import Collector
+from loggly import LogglyCollector
+
+##########################################################################
+
+
+def run_only_if_requirements_present(func):
+    try:
+        import requests
+    except ImportError:
+        requests = None
+    try:
+        import futures
+    except ImportError:
+        futures = None
+    try:
+        import bunch
+    except ImportError:
+        bunch = None
+    return run_only(func,
+                    lambda: all((module is not None
+                                for module in [requests, futures, bunch])))
+
+
+class TestLogglyCollector(CollectorTestCase):
+    TEST_CONFIG = {
+        'interval': 10,
+        'metric': {
+            'testmetric': {
+                'subdomain': 'mocksub',
+                'username': 'mockuser',
+                'password': 'mockpass',
+                'field': 'mock.field',
+                'query': 'mock: query'
+            },
+            'countmetric': {
+                'subdomain': 'mocksub',
+                'username': 'mockuser',
+                'password': 'mockpass',
+                'field': 'mock.field',
+                'count': 'true',
+                'query': 'mock: count_query'
+            },
+            'badmetric': {
+                'subdomain': 'mocksub',
+                'username': 'mockuser',
+                'password': 'mockpass',
+                'field': 'mock.field',
+                'count': 'true',
+                'query': 'mock: bad'
+            },
+            'unauthorizedmetric': {
+                'subdomain': 'mocksub',
+                'username': 'mockuser',
+                'password': 'mockpass',
+                'field': 'mock.field',
+                'count': 'true',
+                'query': 'mock: unauthorized'
+            },
+            'forbiddenmetric': {
+                'subdomain': 'mocksub',
+                'username': 'mockuser',
+                'password': 'mockpass',
+                'field': 'mock.field',
+                'count': 'true',
+                'query': 'mock: forbidden'
+            },
+            'gonemetric': {
+                'subdomain': 'mocksub',
+                'username': 'mockuser',
+                'password': 'mockpass',
+                'field': 'mock.field',
+                'count': 'true',
+                'query': 'mock: gone'
+            },
+            'serverrmetric': {
+                'subdomain': 'mocksub',
+                'username': 'mockuser',
+                'password': 'mockpass',
+                'field': 'mock.field',
+                'count': 'true',
+                'query': 'mock: serverr'
```

Comment:
haha nice, lets make is `server error` since it doesn't seem to abbreviate any of the others.